Update Ballerina Syntax Guide to include missing variables (#245)
wso2/docs-choreo-dev: en/docs/references/ballerina-expression-syntax.md

  +950.930
  ```

### decimal

> The `decimal` type supports integer values with up to 34 decimal points and supports the same arithmetic operators as `int`. When entering literals, you can use `d` as the suffix.

Reviewer: IMO, it's more useful to say something like (quoting from the spec)

```suggestion
The `decimal` type corresponds to a subset of IEEE 754-2008 128-bit decimal (radix 10) floating point numbers.

It supports the same arithmetic operations as `int` and `float`. When entering literals, you can use `d` as the suffix can be used to indicate that the value is a `decimal` value.
```

> The `decimal` type supports integer values with up to 34 decimal points, and supports the same arithmetic operators as `int`. When entering literals, you can use `d` as the suffix.

Reviewer 2: ```suggestion
The `decimal` type supports integer values with up to 34 decimal points and supports the same arithmetic operators as `int`. When entering literals, you can use `d` as the suffix.
```

> This type does not support `infinity`, `NaN` or negative zero.

Reviewer 2: ```suggestion
This type does not support `infinity`, `NaN`, or negative zero.
```

> This type does not support `infinity`, `NaN`, or negative zero.

Reviewer: Why are we saying negative zero is not supported?

Reviewer: Spec says "NaN, infinities and **subnormals** are not supported".

Author: ```suggestion
This type does not support `infinity`, `NaN`, or subnormals.
```

> The `decimal` type is a separate basic type and therefore counts as `anydata`.

Reviewer: The fact that `decimal` is a separate basic type doesn't affect it being a subtype of `anydata`.

```suggestion
The `decimal` type is a separate basic type and is a subtype of `anydata`.
```

> The `decimal` type is a separate basic type, and therefore counts as `anydata`.

Reviewer 2: ```suggestion
The `decimal` type is a separate basic type and therefore counts as `anydata`.
```

###### Examples:

To initialize a variable of type `decimal` in Choreo, you can use one of the following expressions:

  ```ballerina
  0.33536363363636391745284629571635428
  
  -8593.9926666666666666663333333333
  

>   +950.930999999992727272777777777774

Reviewer: Shall we add an example with the `d` suffix also?

Author: Done

"Hello!"
```

### error

> The `error` type supports error values that include a string message. The expressions to initialize this variable type needs to be entered in the `error("<ERROR_STRING>")` format.

Reviewer: ```suggestion
The `error` type represents error values. Error values contain a message of type `string` and optionally a `cause` and additional details about the error. The expressions to initialize a variable of type `error` needs to be entered in the `error("<MESSAGE>")` format.
```

Shouldn't we also include examples of passing the cause and/or details? Aren't these supported yet?

###### Examples:

To initialize a variable of type `error` in Choreo, you can enter an expression as follows:

```ballerina
error("The item requested is not found.")
```

### any

> The any type can represent a variable of any type except the error type. When you do not have prior knowledge of the type of a variable, you can assign it to a variable of type any. Values of these variables can come from dynamic content such as the request and response message references and reference variable types.

Reviewer 2: Code formatting needs to be applied for the term any:
"The `any` type can represent. . . ."

Author: Done

> The `any` type can represent a variable of any type except the error type. When you do not have prior knowledge of the type of a variable, you can assign it to a variable of type any. Values of these variables can come from dynamic content such as the request and response message references and reference variable types.

Reviewer: ```suggestion
The `any` type represents any value that is not an error. When you do not have prior knowledge of the type of a variable, you can assign it to a variable of type `any`. Values assigned to a variable of type `any` can come from dynamic content such as the request and response message references and reference variable types.
```

Reviewer: Do we need to say "When you do not have prior knowledge of the type of a variable, you can assign it to a variable of type `any`"? This isn't quite correct IMO. We will always know the static type.

###### Examples:

To initialize a variable of type `any` in Choreo, you can enter expressions as follows:

    ```ballerina
        The prices are as follows:
  
        123
  
        349.76
    ```
    

### anydata

> The `anydata` type consists of pure values whose basic type is not `error`.

Reviewer 2: Suggestion:
Should we consider replacing "whose" with "where the".

Author: How about "of which the"?

Reviewer 2: +1

> The `anydata` type consists of pure values of which the basic type is not `error`.

Reviewer: Shouldn't we also specify the exact types?

```
  () | boolean | int | float | decimal
    | string | xml
    | anydata[] | map<anydata> | table<map<anydata>>
```

###### Examples:

To initialize a variable of type `anydata` in Choreo, you can enter expressions as follows:

    ```ballerina
        x2 = x1.clone();
        

>         eq = (x1 == x2);

Reviewer: Don't we usually mention only the right hand side expression?

Author: Corrected

```ballerina
check theRevenantJson.cloneWithType(Movie);
```

## XML

> XML is a textual format for representing a string or an XML object. The following topics describe these values.

Reviewer: We don't use the term XML object. Shall we revisit this entire section?

Relevant section in the spec - https://ballerina.io/ballerina-spec/spec.html#XML.

Author: Done

### String values in XML

> Ballerina allows you to cast a string value in XML via variables of type `xml`.

Reviewer: This is also not correct.

Author: Rewrote this section.

###### Examples:

To initialize a variable of type `xml`, you can enter an expression as follows:

```ballerina
    xml `"Dear customer,"`
```

### XML objects

> You can define an XML object to hold XML values.

Reviewer: I think we have to use the proper XML terms here.

Author: Rewrote this section.

###### Examples:

> To declare a JSON object, you can enter an expression as follows:

Reviewer: The example is of an XML element.

Author: Corrected
